### PR TITLE
Task-35725: Hide change password from settings for LDAP users

### PR DESCRIPTION
**Pull request by ahmadmido**
. This PR will assure those only internal exo users can show settings to change password.
. LDAP users can't edit their password.

### CODE REVIEW

File: webapp/portlet/src/main/webapp/user-setting-security/components/UserSettingSecurity.vue [hunk 1]
```diff
@@ -49,6 +52,13 @@ export default {
     });
     document.addEventListener('showSettingsApps', () => this.displayed = true);
     this.skeleton = false;
+    getCurrentUser().then(data => {
+      if(data && data.isInternal){
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/user-setting-security/components/UserSettingSecurity.vue]: this.canChangePassword= data && (data.isInternal || exo.portal.allow.change.external.password == true) ;

File: webapp/portlet/src/main/webapp/user-setting-security/components/UserSettingSecurity.vue [hunk 2]
```diff
@@ -31,13 +31,16 @@
...
+import {getCurrentUser} from '../userSettingSecurity.js';
+
 export default {
   data: () => ({
     id: `Security${parseInt(Math.random() * 10000)
       .toString()
       .toString()}`,
     displayed: true,
     displayDetails: false,
+    isInternal: false,
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/user-setting-security/components/UserSettingSecurity.vue]:
```suggestion
    canChangePassword: true,
```

File: webapp/portlet/src/main/webapp/user-setting-security/components/UserSettingSecurity.vue [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <v-app v-if="displayed">
+  <v-app v-if="displayed && isInternal">
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/user-setting-security/components/UserSettingSecurity.vue]:
```suggestion
  <v-app v-if="displayed && canChangePassword">
```

File: component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java [hunk 4]
```diff
@@ -183,6 +186,11 @@ public static ProfileEntity buildEntityProfile(Profile profile, String restPath,
     userEntity.setAboutMe((String) profile.getProperty(Profile.ABOUT_ME));
     userEntity.setAvatar(profile.getAvatarUrl());
     userEntity.setBanner(profile.getBannerUrl());
+    try {
+      userEntity.setIsInternal(isInternal(profile.getIdentity().getRemoteId()));
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java]:
```suggestion
      userEntity.setIsInternal(isInternal(userEntity.getUsername());
```

File: component/service/src/main/java/org/exoplatform/social/rest/entity/ProfileEntity.java [hunk 5]
```diff
@@ -64,6 +64,8 @@
 
   public static final String ENABLED          = "enabled";
 
+  public static final String ISINTERNAL          = "isInternal";
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/entity/ProfileEntity.java]:
```suggestion
  public static final String IS_INTERNAL          = "isInternal";
```

File: component/service/src/main/java/org/exoplatform/social/rest/entity/ProfileEntity.java [hunk 6]
```diff
@@ -227,6 +229,17 @@ public ProfileEntity setIsMember(boolean isMember) {
     return this;
   }
 
+  public ProfileEntity setIsInternal(boolean IsInternal) {
+    DataEntity dataEntity = getDataEntity();
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/entity/ProfileEntity.java]:
```suggestion
    setProperty("IS_INTERNAL", IsInternal);
```

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: webapp/portlet/src/main/webapp/user-setting-security/components/UserSettingSecurity.vue [hunk 7]
```diff
@@ -50,6 +60,18 @@ export default {
       }
     });
     document.addEventListener('showSettingsApps', () => this.displayed = true);
+    this.skeleton = false;
+    getCurrentUser().then(data => {
+      if(data && data.IS_INTERNAL){
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/user-setting-security/components/UserSettingSecurity.vue]:
```suggestion
     this.canChangePassword= data && (data.isInternal || this.allowChangeExternalPassword ==='true') ;
```
try this instead

File: webapp/portlet/src/main/webapp/user-setting-security/main.js [hunk 8]
```diff
@@ -24,21 +24,23 @@ const url = `${eXo.env.portal.context}/${eXo.env.portal.rest}/i18n/bundle/locale
 
 const appId = 'UserSettingSecurity';
 
-document.dispatchEvent(new CustomEvent('displayTopBarLoading'));
 
-export function init(ssoEnabled) {
-  if (ssoEnabled) {
+export function init(ssoEnabled, allowChangeExternalPassword) {
     document.dispatchEvent(new CustomEvent('hideTopBarLoading'));
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/user-setting-security/main.js:29]: why this is modified ?

File: webapp/portlet/src/main/webapp/user-setting-security/main.js [hunk 9]
```diff
@@ -24,21 +24,23 @@ const url = `${eXo.env.portal.context}/${eXo.env.portal.rest}/i18n/bundle/locale
...
       new Vue({
+
+        data: () => ({
+          allowChangeExternalPassword: allowChangeExternalPassword,
+        }),
         mounted() {
           document.dispatchEvent(new CustomEvent('hideTopBarLoading'));
         },
-        template: `<user-setting-security id="${appId}" v-cacheable />`,
+        template: `<user-setting-security id="${appId}" v-cacheable :allowChangeExternalPassword="allowChangeExternalPassword" />`,
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/user-setting-security/main.js]:
```suggestion
        template: `<user-setting-security id="${appId}" allowChangeExternalPassword="${allowChangeExternalPassword}" v-cacheable />`,
```

File: webapp/portlet/src/main/webapp/user-setting-security/main.js [hunk 10]
```diff
@@ -24,21 +24,23 @@ const url = `${eXo.env.portal.context}/${eXo.env.portal.rest}/i18n/bundle/locale
...
+export function init(ssoEnabled, allowChangeExternalPassword) {
     document.dispatchEvent(new CustomEvent('hideTopBarLoading'));
   } else {
     exoi18n.loadLanguageAsync(lang, url).then(i18n => {
       const appElement = document.createElement('div');
       appElement.id = appId;
 
       new Vue({
+
+        data: () => ({
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/user-setting-security/main.js]: not needed

File: webapp/portlet/src/main/webapp/user-setting-security/components/UserSettingSecurity.vue [hunk 11]
```diff
@@ -30,13 +30,23 @@
...
+    },
+  },
   data: () => ({
     id: `Security${parseInt(Math.random() * 10000)
       .toString()
       .toString()}`,
     displayed: true,
     displayDetails: false,
+    canChangePassword: true,
+    skeleton: true,
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/user-setting-security/components/UserSettingSecurity.vue:49]: why skeleton added ?

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: webapp/portlet/src/main/webapp/user-setting-security/main.js (hunk 9 above)
Comment (ahmadmido) [reply webapp/portlet/src/main/webapp/user-setting-security/main.js]: Done !

**Comment by azayati** [comment]
I have suggested another solution to @ahmadmido, to get the user from conversation state from the jsp file, check the condition of canChangePassword which will be (!ssoEnabled && (internalUser || exo.portal.allow.change.external.password == true), that way we can decide if we keep the profileentity modification or not.
@ahamdi can you give your feedback on that ?

**Review by azayati — APPROVED** [review]
LGTM

File: webapp/portlet/src/main/webapp/WEB-INF/jsp/userSettingSecurity.jsp [hunk 12]
```diff
@@ -1,14 +1,25 @@
...
-      );
-    </script>
+<%@ page import="org.exoplatform.services.organization.OrganizationService" %>
+<%@ page import="org.exoplatform.commons.utils.CommonsUtils"%>
+
+<%
+  OrganizationService organizationService = CommonsUtils.getService(OrganizationService.class);
+  boolean isInternalUser = organizationService.getUserHandler().findUserByName(request.getRemoteUser()).isInternalStore();
+  boolean ssoEnabled = SSOUtils.isSSOEnabled();
+  String allowChangeExternalPassword = System.getProperty("exo.portal.allow.change.external.password");
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/WEB-INF/jsp/userSettingSecurity.jsp:9]:
```suggestion
  boolean allowChangeExternalPassword = System.getProperty("exo.portal.allow.change.external.password", "false").equals("true");
```

File: webapp/portlet/src/main/webapp/WEB-INF/jsp/userSettingSecurity.jsp [hunk 13]
```diff
@@ -1,14 +1,25 @@
...
+%>
+
+<% if (!ssoEnabled && (isInternalUser || allowChangeExternalPassword.equals("true"))) { %>
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/WEB-INF/jsp/userSettingSecurity.jsp:12]:
```suggestion
<% if (!ssoEnabled && (isInternalUser || allowChangeExternalPassword )) { %>
```

**Review by azayati — CHANGES_REQUESTED** [review]
Updated

**Comment by azayati** [comment]
To be tested @ahmadmido

**Review by azayati — DISMISSED** [review]
LGTM

**Review by azayati — APPROVED** [review]
LGTM after QA validation

**Pull request merged**